ast/TreeEquality.cc

```diff
@@ -281,7 +281,12 @@ bool structurallyEqual(const core::GlobalState &gs, const void *avoid, const Tag
                 return false;
             }
             if (a->original && b->original) {
-                return structurallyEqual(gs, avoid, a->original, b->original, file);
+                auto &alit = *a->original;
+                auto &blit = *b->original;
+                if (alit.cnst != blit.cnst) {
+                    return false;
+                }
```

Comment:
Shouldn't this check if `a->original` and `b->original` are `nullptr` before dereferencing them?

Reply:
We are in a `a->original && b->original` codepath, so this is checked already!

Reply:
How did I miss that! For some reason I was reading this block as being directly under a tag case, but it's definitely not 😅